### ISSUE TITLE
Cleanup on bootstrap fail

### ISSUE THREAD

**Issue by thomastaylor312** [deislabs/krustlet#369]
Right now, if the bootstrap process fails part of the way through, it leaves a bunch of invalid resources around in the cluster (like the old CSR). As a new user, it would be helpful if I didn't have to clean these up manually after a bootstrap fail.

Requirements:
- Clean up auth certificate CSR on failure

Questions:
- Should we clean up anything else? Cleaning up local config sounds dangerous, as it still could be valid and would force bootstrapping again

**Comment by thomastaylor312** [comment]
This would probably be a good first issue once we answer the outstanding question

**Comment by itowlson** [comment]
If "bootstrap fail" refers to the bootstrapping that happens within the kubelet, then we should delete only the CSR; leave all the script stuff intact.  The user should be able to run the kubelet again and re-attempt the CSR creation and approval using the existing bootstrap config.

If `bootstrap.sh` fails then I think the user can just run it again - there won't be any conflicts there will there?  So the CSRs are the only things that need to be dealt to.

**Comment by thomastaylor312** [comment]
Yep, I believe so. Just wanted to check so we didn't do anything stupid

**Comment by VishnuJin** [comment]
hey guys, I would like to work on this enhancement, any suggestions please !!

**Comment by thomastaylor312** [comment]
Just make sure to implement the point under `Requirements` in the issue above!

**Comment by VishnuJin** [comment]
hi @thomastaylor312 sorry  I was kinda occupied for the past couple of months,
gave this feature a try #705 please let me know if it's the right approach or I'm completely off track :)
